app/Services/OfferService.php

```
@@ -0,0 +1,120 @@
+
+    private function processImages(Offer $offer, ?array $images = null): void
+    {
+        if ($images) {
+            foreach ($images as $image) {
+                $randomString = substr(str_shuffle('abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789'), 0, 8);
+
+                $extension = $image->getClientOriginalExtension();
+
+                $fileName = "gearly-{$offer->id}-{$randomString}.{$extension}";
+
+                $media = $offer->addMedia($image)
```

PHPMD (2.15.0)

Avoid unused local variables such as '$media'. (Unused Code Rules)

(UnusedLocalVariable)